ui/component/or-asset-tree/src/index.ts

```diff
@@ -575,30 +738,215 @@ export class OrAssetTree extends subscribe(manager)(LitElement) {
+
+        if (e.composedPath()) {
+            value = ((e.composedPath()[0] as HTMLInputElement).value) || undefined;
+        }
+
+        this._shouldEnableAttrType(value);
+    }
+
+    protected _shouldEnableAttrType(value: string | undefined): void {
+        if (value) {
+            this._attributeValueFilter.disabled = false;
+        } else {
+            this._attributeValueFilter.disabled = true;
+        }
+    }
+
+    protected applySettingFields(filter: OrAssetTreeFilter): OrAssetTreeFilter {
+        if ( this._assetTypeFilter ) {
+            filter.assetType = [ this._assetTypeFilter ];
+        } else {
+            filter.assetType = [];
+        }
+
+        if ( this._attributeNameFilter.value ) {
+            filter.attribute = [ this._attributeNameFilter.value ];
+        } else {
+            filter.attribute = [];
+        }
+
+        if ( this._attributeNameFilter.value && this._attributeValueFilter.value ) {
+            const attributeValueValue: string = this._attributeValueFilter.value.replace(' ', '');
+            filter.attributeValue = [ attributeValueValue ];
+        } else {
+            filter.attributeValue = [];
+        }
+
+        return filter;
+    }
+
+    protected _filterFromSettings(): void {
+        let filterFromSearchInput: OrAssetTreeFilter = this.parseFromInputFilter();
+
+        let filterFromSearchInputWithSettings: OrAssetTreeFilter = this.applySettingFields(filterFromSearchInput);
+
+        this._filter = filterFromSearchInputWithSettings;
+
+        let newFilterForSearchInput: string = this.formatFilter(this._filter);
+
+        if (newFilterForSearchInput) {
+            this._clearIconContainer.classList.add("visible");
+        } else {
+            this._clearIconContainer.classList.remove("visible");
+        }
+
+        this._filterInput.value = newFilterForSearchInput;
+
+        this._filterSetting.classList.remove("visible");
+
+        this._doFiltering();
+    }
+
+    protected _onFilterInputEven(e: KeyboardEvent) {
         let value: string | undefined;
 
         if (e.composedPath()) {
             value = ((e.composedPath()[0] as HTMLInputElement).value) || undefined;
 
             if (value) {
-                this._clearIcon.classList.add("visible");
+                this._clearIconContainer.classList.add("visible");
             } else {
-                this._clearIcon.classList.remove("visible");
+                this._clearIconContainer.classList.remove("visible");
             }
         }
 
+        this._onFilterInput(value, false);
+    }
+
+    protected _onFilterInput(newValue: string | undefined, force: boolean): void {
+        if (newValue) {
+            this._clearIconContainer.classList.add("visible");
+        } else {
+            this._clearIconContainer.classList.remove("visible");
+        }
+
+        let currentFilter: OrAssetTreeFilter = this.parseFromInputFilter(newValue);
+
+        if (this._filter.asset === currentFilter.asset &&
```

Comment:
There is a util helper method for object deepEquals checking

Reply:
You're right, I moved to Util.equals